src/main/cpp/chassis/driveStates/DriveToAmp.cpp

```diff
@@ -0,0 +1,101 @@
+//====================================================================================================================================================
+// Copyright 2024 Lake Orion Robotics FIRST Team 302
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+// OR OTHER DEALINGS IN THE SOFTWARE.
+//=====================================================================================================================================================
+
+// Team302 includes
+#include "chassis/driveStates/DriveToAmp.h"
+#include "chassis/ChassisConfigMgr.h"
+
+#include "utils/FMSData.h"
+
+// third party includes
+#include "pathplanner/lib/path/PathPlannerPath.h"
+
+DriveToAmp::DriveToAmp() : m_chasis(nullptr)
+{
+    auto config = ChassisConfigMgr::GetInstance()->GetCurrentConfig();
+    m_chasis = config != nullptr ? config->GetSwerveChassis() : nullptr;
+    m_dragonVision = DragonVision::GetDragonVision();
+}
+
+pathplanner::PathPlannerTrajectory DriveToAmp::CreateDriveToAmpPath()
+{
+    frc::DriverStation::Alliance allianceColor = FMSData::GetInstance()->GetAllianceColor();
+
+    m_currentPose2d = m_chasis->GetPose();
```

Comment:
Don't think current pose should be a class attribute as it is changing every time this gets called